src/client/views/components/todoItem.jsx

```diff
@@ -116,6 +118,17 @@ export default React.createClass({
         });
     },
 
+    makeElementFullyVisible: function () {
+        const { top: elementTop, bottom: elementBottom } = ReactDOM.findDOMNode(this).getBoundingClientRect();
+        const isVisible = (elementTop >= 0) && (elementBottom <= window.innerHeight);
+
+        if (!isVisible) {
+            setTimeout(() => {
+                window.scrollBy(0, elementBottom - elementTop);
+            }, 100);
```

Comment:
OK, I think I have a way around this, and it's the `transitionend` event.
http://www.chloechen.io/react-animation-done-in-two-ways/

Basically, this method wouldn't be called by a component mounting, but instead the transition/animation completing. You would want to make sure this transition was only set once though and when it's a new item.

Otherwise, if you just want to move on from this for now (understandable), I'd suggest setting the `100` to `150` (the actual time for the height animation) as constant at the top of this file (around line 4) and wrapping this whole method call in the `setTimeout` instead - so that line 30 would become `setTimeout(this.makeElementFullyVisible, HEIGHT_ANIMATION_TIME);`